test/test_torch.py

```diff
@@ -17228,6 +17228,122 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    # Use double copysign to verify the correctnes of 0.0 and -0.0,
+    # since self.assertEqual(0.0 == -0.0)
+    def test_copysign_bool(self, device):
+        # Result is promoted to float
+        # Only 0 and 1
+        x = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.bool)
+        y = torch.tensor([0, 1, 0, 1], device=device, dtype=torch.bool)
+        expected = torch.tensor([0, 0, 1, 1], device=device, dtype=torch.float)
+        self.assertEqual(torch.copysign(torch.tensor(1.0), torch.copysign(x, y)),
+                         torch.copysign(torch.tensor(1.0), expected))
+
+    @dtypes(torch.half, torch.float, torch.double)
+    def test_copysign_floating(self, device, dtype):
```

Comment:
I think these special cases can be enumerated programmatically as part of the test case below.

Reply:
Also, I think these tests need to add cases for inf, -inf and nan.